### ISSUE TITLE
Unknown config fields: auth.hook.send_email.secret

### ISSUE THREAD

**Issue by florian-lefebvre** [supabase/supabase#28784]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

In my `config.toml`, I have the following config for auth hooks ([following the docs](https://supabase.com/docs/guides/auth/auth-hooks?queryGroups=language&language=http#developing)):

```toml
[auth.hook.send_email]
enabled = true
uri = "http://localhost:3000/my-api/send-email"
secret = "env(SUPABASE_SEND_EMAIL_HOOK_SECRET)"
```

However, restarting the cli (`supabase stop && supabase start` logs the following warning:

```
Unknown config fields: [auth.hook.send_email.secret auth.hook.send_email.secret auth.hook.send_email.secret]
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Use supabase CLI
2. Set your `config.toml` to the example above
3. Restart the CLI
4. See the warning in the terminal

## Expected behavior

The warning should not appear. I don't know yet if it's a false positive or if the secret is not taken into account at all.

- EDIT 1: does not seem to be taken into account
- EDIT 2: looks like the field is called `secrets`, not `secret` (https://github.com/supabase/cli/blob/fe3623ae429cd667fe7de494424fe38c9b7e3869/api/beta.yaml#L3861) so that's a docs issue most likely

## System information

- OS: Windows 11
- Version of supabase: 1.190.0
- Version of supabase-js: 2.45.1
- Version of Node.js: 20.11.1

**Comment by encima** [comment]
Thanks for opening, it looks like a docs and/or implementation issue for sure. Singular `secret` makes more sense so maybe the CLI needs to be corrected for this.

We will keep this in the main repo (instead of CLI) for now and see what the docs team think!

**Comment by florian-lefebvre** [comment]
Yeah I agree `secret` makes more sense. Also, I think it would be worth clarifying docs a bit about the values:
- In `config.toml` `secrets`, the value must be `v1,whsec_<base64_encoded>`
- In the webhook, the secret to use with the sandard webhook library is actually just `<base64_encoded>`

Since in config.toml, you cannot use a partial env var (eg. `v1,whsec_env(VAR)`), I think it would make sense to use the following as an example:

```init
# .env
SEND_EMAIL_SMS_SECRET=v1,whsec_<base64_encoded>
```

```toml
# config.toml

[auth.hook.send_email]
# ...
secret = "env(SEND_EMAIL_SMS_SECRET)"
```

```js
const SEND_EMAIL_SMS_SECRET = getFromEnv()
new Webhook(SEND_EMAIL_SMS_SECRET.replace("v1,whsec_", "")) // I'm sure .slice would be more efficient but less understandable
```

**Comment by encima** [comment]
@florian-lefebvre fully agree, if you want to make a PR for the docs, I can make a PR for the CLI (after confirming with the team)? Or you are welcome to do both if you like? Let me know!

**Comment by encima** [comment]
Let me check with the CLI/auth team about their thoughts and get them to weigh in here

**Comment by sweatybridge** [comment]
> In the webhook, the secret to use with the sandard webhook library is actually just <base64_encoded>

@florian-lefebvre could you elaborate which standard webhook library you are referring to? AFAIK both hosted dashboard and cli config.toml expects hook secret to be prefixed with `v1,whsec_` currently.

As @J0 mentioned to me, plural `secrets` is used because it accepts a comma separated list of secret values. So perhaps we only need to update the docs to use the correct field.

**Comment by florian-lefebvre** [comment]
https://supabase.com/docs/guides/auth/auth-hooks?queryGroups=language&language=http#security-model

**Comment by florian-lefebvre** [comment]
> could you elaborate which standard webhook library you are referring to? AFAIK both hosted dashboard and cli config.toml expects hook secret to be prefixed with v1,whsec_ currently.

I'm talking about https://supabase.com/docs/guides/auth/auth-hooks?queryGroups=language&language=http#security-model. I just found docs unclear about what values were expected. The supabase config indeed requires the `v1,whsec_` prefix but the `Webhook` class from the `standardwebhooks` package (as recommended in this section of the docs) requires the secret without the prefix.

> plural secrets is used because it accepts a comma separated list of secret values. So perhaps we only need to update the docs to use the correct field.

yeah in this case it makes sense, i can update it

**Comment by J0** [comment]
>As @J0 mentioned to me, plural secrets is used because it accepts a comma separated list of secret values. So perhaps we only need to update the docs to use the correct field.

Yes, that's right. 

>yeah in this case it makes sense, i can update it

Thank you! Let us know if there are any issues here

**Comment by florian-lefebvre** [comment]
Just need to submit a docs PR, keeping this issue to track

**Comment by Bartek532** [comment]
Is it normal that I'm receiving data without encoding in local development?
